flask_appbuilder/security/decorators.py

```diff
@@ -2,12 +2,53 @@
 import functools
 
 from flask import flash, redirect, url_for, make_response, jsonify, request
+from flask_jwt_extended import verify_jwt_in_request,  get_jwt_identity
+from flask_login import login_user
 from .._compat import as_unicode
 from ..const import LOGMSG_ERR_SEC_ACCESS_DENIED, FLAMSG_ERR_SEC_ACCESS_DENIED, PERMISSION_PREFIX
 
 log = logging.getLogger(__name__)
 
-        
+
+def protect(f):
+    """
+        Use this decorator to enable granular security permissions
+        to your API methods (BaseApi and child classes).
+        Permissions will be associated to a role, and roles are associated to users.
+
+        By default the permission's name is the methods name.
+    """
+    if hasattr(f, '_permission_name'):
+        permission_str = f._permission_name
+    else:
+        permission_str = f.__name__
+
+    def wraps(self, *args, **kwargs):
+        permission_str = "{}{}".format(PERMISSION_PREFIX, f._permission_name)
+        if self.appbuilder.sm.is_item_public(
+                permission_str,
+                self.__class__.__name__
+        ):
+            return f(self, *args, **kwargs)
+        verify_jwt_in_request()
+        login_user(self.appbuilder.sm.get_user_by_id(int(get_jwt_identity())))
```

Comment:
Probably not the right place for this, but we're assuming here that users will be identified by a unique `int` (I'm assuming a PKEY on the user table). Would it be a big lift to allow for strings (UUIDs) as user ids? Again, this isn't really the best place for this, but in terms of "best practices" in security, using a one-up counter for security-related ids is usually frowned upon.

Reply:
I see what you mean, and yes, this is probably not the best place. but let's talk about it